mslib/mswms/wms.py

```diff
@@ -329,209 +344,212 @@ def produce_plot(self, query, mode):
+                    return self.create_service_exception(
+                        code="MissingDimensionValue",
+                        text="TIME not specified",
+                        version=version)
 
-            # Bounding box (num interp. points, p_bot, num labels, p_top).
-            try:
-                bbox = [float(v) for v in query.get("BBOX", "101,1050,10,180").split(",")]
-            except ValueError:
-                return self.create_service_exception(text=f"Invalid BBOX: {query.get('BBOX')}", version=version)
+                # Check if the requested coordinate system is supported.
+                if not self.hsec_layer_registry[dataset][layer].support_epsg_code(crs):
+                    return self.create_service_exception(
+                        code="InvalidSRS",
+                        text=f"The requested CRS '{crs}' is not supported.",
+                        version=version)
 
-            plot_driver = self.vsec_drivers[dataset]
-            try:
-                plot_driver.set_plot_parameters(plot_object=self.vsec_layer_registry[dataset][layer],
-                                                vsec_path=path,
-                                                vsec_numpoints=bbox[0],
-                                                vsec_path_connection="greatcircle",
-                                                vsec_numlabels=bbox[2],
-                                                init_time=init_time,
-                                                valid_time=valid_time,
-                                                style=style,
-                                                bbox=bbox,
-                                                figsize=figsize,
-                                                noframe=noframe,
-                                                transparent=transparent,
-                                                return_format=return_format)
-                image = plot_driver.plot()
-            except (IOError, ValueError) as ex:
-                logging.error("ERROR: %s %s", type(ex), ex)
-                msg = "The data corresponding to your request is not available. Please check the " \
-                      "times and/or path you have specified.\n\n" \
-                      f"Error message: {ex}"
-                return self.create_service_exception(text=msg, version=version)
+                # Bounding box.
+                try:
+                    if is_yx:
+                        bbox = [float(v) for v in query.get('BBOX', '-90,-180,90,180').split(',')]
+                        bbox = (bbox[1], bbox[0], bbox[3], bbox[2])
+                    else:
+                        bbox = [float(v) for v in query.get('BBOX', '-180,-90,180,90').split(',')]
+
+                except ValueError:
+                    return self.create_service_exception(text=f"Invalid BBOX: {query.get('BBOX')}", version=version)
```

Comment:
in the past there was also an IOError option and a logging, IOError won't happen without IO ;)
logging.error("ERROR: %s %s", type(ex), ex)